website/docs/r/log_analytics_workspace_table.html.markdown

```diff
@@ -41,9 +41,15 @@ The following arguments are supported:
 
 * `workspace_id` - (Required) The object ID of the Log Analytics Workspace that contains the table.
 
-* `retention_in_days` - (Required) The table's retention in days. Possible values are either 7 (Free Tier only) or range between 30 and 730.
+* `plan` - (Optional) Specify the system how to handle and charge the logs ingested to the table. Possible values are `Analytics` and `Basic. Defaults to `Analytics`.
 
--> **Note:** retention_in_days will revert back to the value of azurerm_log_analytics_workspace retention_in_days when a azurerm_log_analytics_workspace_table is deleted.
+-> **Note:** The `name` of tables currently support `Basic` plan can be found in [this article](https://learn.microsoft.com/en-us/azure/azure-monitor/logs/basic-logs-configure?tabs=portal-1#supported-tables) for further details.
+
+* `retention_in_days` - (Optional) The table's retention in days. Possible values are either 7 (Free Tier only) or range between 30 and 730.
+
+-> **Note:** `retention_in_days` will revert back to the value of azurerm_log_analytics_workspace retention_in_days when a azurerm_log_analytics_workspace_table is deleted.
+
+-> **Note:** The `retention_in_days` cannot be specified when the `plan` is `Basic` because it is immutable.
```

Comment:
"because the retention is fixed to 8 days" would give a better information to the user.

Reply:
Same as above.

Reply:
I believe it is safe to mention the 8 days. The product is GA and this value is mentioned multiple times in the documentation:

https://learn.microsoft.com/en-us/azure/azure-monitor/logs/basic-logs-configure?tabs=portal-1#compare-the-basic-and-analytics-log-data-plans

![Screenshot 2023-12-29 at 10 35 02](https://github.com/hashicorp/terraform-provider-azurerm/assets/789701/6a883db4-372b-4604-b9e9-b56a03227c60)

Reply:
Thanks for the explanation. The code has been updated.